backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/SchedulingManager.java

```diff
@@ -431,7 +432,8 @@ private Map<Guid, Guid> schedule(Cluster cluster,
                         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, NumaNodeMemoryConsumption::merge));
                 updateHostNumaNodes(host, numaConsumption);
 
-
+                // Sorting the VMs to get better dedicated CPU pinning.
+                vmsNotOnHost.sort(new VmsCpuPinningPolicyComparator());
```

Comment:
I am afraid that if we omit the sorting on the `canSchedule` we might filter out some hosts that are able to run the vm group. I would suggest to sort the vms in the `groupVms` method - that way we could be sure that it is used sorted thoughtout the `SchedulingManager`.